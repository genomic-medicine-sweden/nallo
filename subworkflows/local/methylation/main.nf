include { MODKIT_PILEUP    } from '../../../modules/nf-core/modkit/pileup/main'
include { TABIX_BGZIPTABIX } from '../../../modules/nf-core/tabix/bgziptabix/main'

workflow METHYLATION {

    take:
    ch_bam_bai             // channel: [ val(meta), bam, bai ]
    ch_fasta               // channel: [ val(meta), fasta ]
    ch_bed                 // channel: [ val(meta), bed ]

    main:
    ch_versions = Channel.empty()

    // Performs pileups per haplotype if the phasing workflow is on, set in config
    MODKIT_PILEUP (ch_bam_bai, ch_fasta, ch_bed)
    ch_versions = ch_versions.mix(MODKIT_PILEUP.out.versions)


    MODKIT_PILEUP.out.bed
        .transpose()
        .set { ch_bgzip_modkit_pileup_in }

    TABIX_BGZIPTABIX ( ch_bgzip_modkit_pileup_in )
    ch_versions = ch_versions.mix(TABIX_BGZIPTABIX.out.versions)

    emit:
    bed      = TABIX_BGZIPTABIX.out.gz_tbi.map { meta, bed, _tbi -> [ meta, bed ] } // channel: [ val(meta), path(bed) ]
    tbi      = TABIX_BGZIPTABIX.out.gz_tbi.map { meta, _bed, tbi -> [ meta, tbi ] } // channel: [ val(meta), path(tbi) ]
    versions = ch_versions // channel: [ versions.yml ]
}

