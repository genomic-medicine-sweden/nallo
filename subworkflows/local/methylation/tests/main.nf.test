nextflow_workflow {

    name "Test Workflow METHYLATION"
    script "../main.nf"
    config "./nextflow.config"
    workflow "METHYLATION"

    setup {
        run("PREPARE_GENOME") {
            script "../../prepare_genome.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'hg38' ],
                    file(params.pipelines_testdata_base_path + 'reference/hg38.test.fa.gz', checkIfExists: true)
                ])
                input[1] = true
                input[2] = Channel.empty()
                input[3] = false
                """
            }
        }
        run("MINIMAP2_ALIGN") {
            script "../../../../modules/nf-core/minimap2/align/main.nf"
            process {
                """
                input[0] = [
                    [ id: 'test', num_intervals:1 ],
                    file(params.pipelines_testdata_base_path + 'testdata/HG002_PacBio_Revio.fastq.gz', checkIfExists: true)
                ]
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = true
                input[3] = 'csi'
                input[4] = false
                input[5] = false
                """
            }
        }
        run("SHORT_VARIANT_CALLING") {
            script "../../short_variant_calling/main.nf"
            process {
                """
                input[0] = MINIMAP2_ALIGN.out.bam
                            .join(MINIMAP2_ALIGN.out.index)
                            .join(Channel.of([
                                [ id: 'test', num_intervals:1 ],
                                file(params.pipelines_testdata_base_path + 'reference/test_data.bed', checkIfExists: true)
                            ]))
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = PREPARE_GENOME.out.fai
                input[3] = [[],[]]
                input[4] = [[],[]]
            """
            }
        }
        run("PHASING") {
            script "../../phasing.nf"
            process {
                """
                input[0] = SHORT_VARIANT_CALLING.out.snp_calls_vcf
                input[1] = SHORT_VARIANT_CALLING.out.snp_calls_tbi
                input[2] = MINIMAP2_ALIGN.out.bam
                            .join(MINIMAP2_ALIGN.out.index)
                            .map { meta, bam, bai ->
                                [ [ id : meta.id ], bam, bai ]
                            }
                input[3] = PREPARE_GENOME.out.fasta
                input[4] = PREPARE_GENOME.out.fai
                """
            }
        }
    }

    test("1 sample - [] phased") {
        when {
            params {
                skip_phasing = false
            }
            workflow {
                """
                input[0] = PHASING.out.haplotagged_bam_bai
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = PREPARE_GENOME.out.fai
                input[3] = [[],[]]
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("1 sample - [] unphased") {
        when {
            params {
                skip_phasing = true
            }
            workflow {
                """
                input[0] = MINIMAP2_ALIGN.out.bam.join(MINIMAP2_ALIGN.out.index)
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = PREPARE_GENOME.out.fai
                input[3] = [[],[]]
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("1 sample - bed phased") {
        when {
            params {
                skip_phasing = false
            }
            workflow {
                """
                input[0] = PHASING.out.haplotagged_bam_bai
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = PREPARE_GENOME.out.fai
                input[3] = Channel.of('chr20\t2650000\t2660000')
                            .collectFile(name: 'chr20.bed', newLine: true)
                            .map{ file -> [ [ id:'chr20'], file ] }
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("1 sample - bed unphased") {
        when {
            params {
                skip_phasing = true
            }
            workflow {
                """
                input[0] = MINIMAP2_ALIGN.out.bam.join(MINIMAP2_ALIGN.out.index)
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = PREPARE_GENOME.out.fai
                input[3] = Channel.of('chr20\t2650000\t2660000')
                            .collectFile(name: 'chr20.bed', newLine: true)
                            .map{ file -> [ [ id:'chr20'], file ] }
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("1 sample - [] phased - stub") {
        options "-stub"
        when {
            params {
                skip_phasing = false
            }
            workflow {
                """
                input[0] = PHASING.out.haplotagged_bam_bai
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = PREPARE_GENOME.out.fai
                input[3] = [[],[]]
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("1 sample - bed phased - stub") {
        options "-stub"
        when {
            params {
                skip_phasing = false
            }
            workflow {
                """
                input[0] = PHASING.out.haplotagged_bam_bai
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = PREPARE_GENOME.out.fai
                input[3] = Channel.of('chr20\t2650000\t2660000')
                            .collectFile(name: 'chr20.bed', newLine: true)
                            .map{ file -> [ [ id:'chr20'], file ] }
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("1 sample - [] unphased - stub") {
        options "-stub"
        when {
            params {
                skip_phasing = true
            }
            workflow {
                """
                input[0] = MINIMAP2_ALIGN.out.bam.join(MINIMAP2_ALIGN.out.index)
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = PREPARE_GENOME.out.fai
                input[3] = [[],[]]
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("1 sample - bed unphased - stub") {
        options "-stub"
        when {
            params {
                skip_phasing = true
            }
            workflow {
                """
                input[0] = MINIMAP2_ALIGN.out.bam.join(MINIMAP2_ALIGN.out.index)
                input[1] = PREPARE_GENOME.out.fasta
                input[2] = PREPARE_GENOME.out.fai
                input[3] = Channel.of('chr20\t2650000\t2660000')
                            .collectFile(name: 'chr20.bed', newLine: true)
                            .map{ file -> [ [ id:'chr20'], file ] }
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
