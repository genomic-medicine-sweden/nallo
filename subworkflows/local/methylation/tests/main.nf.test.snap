{
    "1 sample - [] phased": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bed.gz:md5,2338b96fe077f6425d13671f4f00621c"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_2.bed.gz:md5,04943736472126e5c908de4ae7edc377"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bed.gz:md5,d3f37c86a25bf976a2f5bc333adcfb27"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bed.gz.tbi:md5,9167ae6a6300da7bc5914f6f8c8614ea"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_2.bed.gz.tbi:md5,8c70bd51c69227e8f5f16fa367a679f6"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bed.gz.tbi:md5,d79a058b21a35f4b869183f75d580f15"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bw:md5,30c07d45dc627179efa5fb30ed4c8e30"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_2.bw:md5,4c74eb63ecb5e872d2b2ae59572d0b2d"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bw:md5,36497f3760683c4371c39a5ec29624fb"
                    ]
                ],
                "3": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ],
                "bed": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bed.gz:md5,2338b96fe077f6425d13671f4f00621c"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_2.bed.gz:md5,04943736472126e5c908de4ae7edc377"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bed.gz:md5,d3f37c86a25bf976a2f5bc333adcfb27"
                    ]
                ],
                "bigwig": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bw:md5,30c07d45dc627179efa5fb30ed4c8e30"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_2.bw:md5,4c74eb63ecb5e872d2b2ae59572d0b2d"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bw:md5,36497f3760683c4371c39a5ec29624fb"
                    ]
                ],
                "tbi": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bed.gz.tbi:md5,9167ae6a6300da7bc5914f6f8c8614ea"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_2.bed.gz.tbi:md5,8c70bd51c69227e8f5f16fa367a679f6"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bed.gz.tbi:md5,d79a058b21a35f4b869183f75d580f15"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-21T15:39:28.719949885"
    },
    "1 sample - bed unphased - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ],
                "bed": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "bigwig": [
                    
                ],
                "tbi": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-21T15:59:47.96185188"
    },
    "1 sample - [] unphased - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ],
                "bed": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "bigwig": [
                    
                ],
                "tbi": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-21T15:59:39.523094893"
    },
    "1 sample - [] phased - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ],
                "bed": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "bigwig": [
                    
                ],
                "tbi": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-09-11T14:03:44.598003624"
    },
    "1 sample - [] unphased": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,8dee9e20f68bdd50e95032ccce8e21b6"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,82b2208891efed61d1a66ecea21b09a7"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bw:md5,98f619ac8dc59465cab7385d5963966f"
                    ]
                ],
                "3": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ],
                "bed": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,8dee9e20f68bdd50e95032ccce8e21b6"
                    ]
                ],
                "bigwig": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bw:md5,98f619ac8dc59465cab7385d5963966f"
                    ]
                ],
                "tbi": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,82b2208891efed61d1a66ecea21b09a7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-21T15:58:52.784354718"
    },
    "1 sample - bed phased": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bed.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bed.gz:md5,11eaa3415b63ef1a4f804a1688241240"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bed.gz.tbi:md5,ec06685d85d15d8f118a789a9ccc42ad"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bed.gz.tbi:md5,f30acad35b312ff0f364a749bb3fe0f6"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bw:md5,e7f25f289f3dd02261ca868084487094"
                    ]
                ],
                "3": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ],
                "bed": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bed.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bed.gz:md5,11eaa3415b63ef1a4f804a1688241240"
                    ]
                ],
                "bigwig": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bw:md5,e7f25f289f3dd02261ca868084487094"
                    ]
                ],
                "tbi": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_1.bed.gz.tbi:md5,ec06685d85d15d8f118a789a9ccc42ad"
                    ],
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup_ungrouped.bed.gz.tbi:md5,f30acad35b312ff0f364a749bb3fe0f6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-21T15:39:51.667873598"
    },
    "1 sample - bed phased - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ],
                "bed": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "bigwig": [
                    
                ],
                "tbi": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-09-11T14:04:09.076767689"
    },
    "1 sample - bed unphased": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,840386053b6a3f50ff8ffbbbba938595"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,c9a0a0d1d170c008224eba2f585c613d"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bw:md5,e7f25f289f3dd02261ca868084487094"
                    ]
                ],
                "3": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ],
                "bed": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz:md5,840386053b6a3f50ff8ffbbbba938595"
                    ]
                ],
                "bigwig": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bw:md5,e7f25f289f3dd02261ca868084487094"
                    ]
                ],
                "tbi": [
                    [
                        {
                            "id": "HG002"
                        },
                        "HG002_modkit_pileup.bed.gz.tbi:md5,c9a0a0d1d170c008224eba2f585c613d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,138cc6e6c6e5328482b1e5c6a31029c8",
                    "versions.yml:md5,357aba3cdaf107b0be7a037803837280",
                    "versions.yml:md5,69dd5e76cc5e61cd0324b7412fc0546b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-21T15:59:14.491504374"
    }
}